person.rb

```diff
@@ -18,3 +29,34 @@ def of_age?
     @age >= 18
   end
 end
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.split.map(&:capitalize).join(' ')
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.size > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
+  end
+end
+
```

Comment:
- [ OPTIONAL ] I highly recommend putting each class in sperate  file to make your project well structured since you are making inheritance on the classes so kindly remove each class to seprate files ✅🔥💯💯

Reply:
thank you I have resolved that